cli-plugins/socket/socket_test.go

```diff
@@ -78,18 +75,21 @@ func checkDirNoPluginSocket(t *testing.T) {
 
 func TestConnectAndWait(t *testing.T) {
 	t.Run("calls cancel func on EOF", func(t *testing.T) {
-		var conn *net.UnixConn
-		listener, err := SetupConn(&conn)
+		listener, conn, err := SetupConn()
 		assert.NilError(t, err, "failed to setup listener")
+		t.Cleanup(func() {
+			if !conn.IsNil() {
+				conn.Close()
+			}
+		})
```

Comment:
TODO: cleanup. I was testing some failures here (or rather flakes). Seems the timeout gets hit _sometimes_...